### ISSUE TITLE
I can not run the gam model with the resampling method none

### ISSUE THREAD

**Issue by elpidiofilho** [topepo/caret#787]
I can not run the gam model with the resampling method none

``` r
Sys.setenv(LANG="EN") 
suppressPackageStartupMessages(library(dplyr))
#> Warning: package 'dplyr' was built under R version 3.4.2
suppressPackageStartupMessages(library(caret))
d = airquality  %>% na.omit() %>%  data.frame()
set.seed(313)
va = caret::createDataPartition(d[,1], p = 0.75, list = F)
train = d[va,]
test  = d[-va,]
tc <- caret::trainControl( method = 'none')
fit1 = caret::train(x = train[,-1],  y = train[,1], method = "gam", trControl = tc)
#> Error: Only one model should be specified in tuneGrid with no resampling
warnings()
#> NULL
sessionInfo()
#> R version 3.4.1 (2017-06-30)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 15063)
#> 
#> Matrix products: default
#> 
#> locale:
#> [1] LC_COLLATE=Portuguese_Brazil.1252  LC_CTYPE=Portuguese_Brazil.1252   
#> [3] LC_MONETARY=Portuguese_Brazil.1252 LC_NUMERIC=C                      
#> [5] LC_TIME=Portuguese_Brazil.1252    
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] caret_6.0-78    ggplot2_2.2.1   lattice_0.20-35 dplyr_0.7.4    
#> 
#> loaded via a namespace (and not attached):
#>  [1] purrr_0.2.4        reshape2_1.4.2     kernlab_0.9-25    
#>  [4] splines_3.4.1      colorspace_1.3-2   stats4_3.4.1      
#>  [7] htmltools_0.3.6    mgcv_1.8-22        yaml_2.1.14       
#> [10] survival_2.41-3    prodlim_1.6.1      rlang_0.1.4       
#> [13] ModelMetrics_1.1.0 withr_2.1.0        glue_1.2.0        
#> [16] bindrcpp_0.2       foreach_1.4.3      bindr_0.1         
#> [19] plyr_1.8.4         dimRed_0.1.0       lava_1.5.1        
#> [22] robustbase_0.92-8  stringr_1.2.0      timeDate_3042.101 
#> [25] munsell_0.4.3      gtable_0.2.0       recipes_0.1.0     
#> [28] codetools_0.2-15   evaluate_0.10.1    knitr_1.17        
#> [31] class_7.3-14       DEoptimR_1.0-8     Rcpp_0.12.13      
#> [34] scales_0.5.0       backports_1.1.1    ipred_0.9-6       
#> [37] CVST_0.2-1         digest_0.6.12      stringi_1.1.5     
#> [40] RcppRoll_0.2.2     ddalpha_1.3.1      grid_3.4.1        
#> [43] rprojroot_1.2      tools_3.4.1        magrittr_1.5      
#> [46] lazyeval_0.2.1     tibble_1.3.4       DRR_0.0.2         
#> [49] pkgconfig_2.0.1    MASS_7.3-47        Matrix_1.2-12     
#> [52] lubridate_1.7.1    gower_0.1.2        assertthat_0.2.0  
#> [55] rmarkdown_1.7      iterators_1.0.8    R6_2.2.2          
#> [58] rpart_4.1-11       sfsmisc_1.1-1      nnet_7.3-12       
#> [61] nlme_3.1-131       compiler_3.4.1
```

**Comment by topepo** [comment]
> `Error: Only one model should be specified in tuneGrid with no resampling`

By default, multiple models are tried. You can specify which one that you want using `tuneGrid`. See `?models` for the list of parameters for that model.

**Comment by dmpe** [comment]
@topepo Hi there, 
sorry but could you maybe improve the error message. It is not really helpful. When I choose `none` method, I would expect to to make 1 train and test run without any tuneGrid parameter....
Please, feel free to correct me if I am wrong (I guess I am). 
thank you.

**Comment by topepo** [comment]
> without any `tuneGrid` parameter.

The model is setup here to have parameters, so at least one (and only one) model should be specified.

**Comment by ries9112** [comment]
I know this is old and has been closed, but I am running into this issue and I'm slightly confused on why it may be happening.

Like @dmpe when training I am using `trControl = trainControl(method = "none")`, and I am confused on where I am specified more than one model.

This does not work:
```
train(sell_price_usd_24h_later ~ . , data = train, 
          method = "xgbLinear",
          na.action = na.pass,
          trControl=trainControl(method="none"))
```

And gives the error: `Error: Only one model should be specified in tuneGrid with no resampling`

But why is it that if I remove `trControl = trainControl(method = "none")` things do indeed work?

This works:
```
train(sell_price_usd_24h_later ~ . , data = train, 
          method = "xgbLinear",
          na.action = na.pass)
```

How does specifying a method on "none" end up specifying more than one model? I am sure there is a reason/rationale, but I can't figure out what that would be based on the discussion that happened here. I am not looking to resolve my particular issue, and I was simply hoping for additional clarity on the past conversation that already took place, but if it is helpful I can provide a reprex as well.

**Comment by ries9112** [comment]
I know this is old and has been closed, but I am running into this issue and I'm slightly confused on why it may be happening.

Like @dmpe when training I am using `trControl = trainControl(method = "none")`, and I am confused on where I am specifying more than one model.

This does not work:
```
train(sell_price_usd_24h_later ~ . , data = train,
          method = "xgbTree",
          na.action = na.pass,
          trControl=trainControl(method="none"))
```

And gives the error: `Error: Only one model should be specified in tuneGrid with no resampling`

But why is it that if I remove `trControl = trainControl(method = "none")` things do indeed work?

This works:
```
train(sell_price_usd_24h_later ~ . , data = train,
          method = "xgbTree",
          na.action = na.pass)
```

Basically I think I am understanding that the error message should be interpreted as "Exactly one model should be specified" and that the `method="none"` parameter seems to be making that number 0, but I am failing to understand why this only happens in some cases. For example, using the method 'xgbLinear' I am not running into any problems when including `trControl=trainControl(method="none"))` which I find odd. I am not looking to resolve my particular issue, and I was simply hoping for additional clarity on the past conversation that already took place, but if it is helpful I can provide a reprex as well.
